presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/SymbolMapper.java

```diff
@@ -66,6 +68,16 @@ public Expression rewriteSymbolReference(SymbolReference node, Void context, Exp
 
     public AggregationNode map(AggregationNode node, PlanNode source)
```

Comment:
Yes, it's used somewhere else, and it's legit.

The whole abstraction is a bit wonky, though. It's used for re-mapping the symbols in the context of certain transformations that need to know how the node will fit in the larger plan. We may need to revisit in the future.